src/postgres/inventory/paginator/PgPaginatorOrderingAttributes.ts

```diff
@@ -66,37 +66,40 @@ implements Paginator.Ordering<TInput, PgClassType.Value, AttributesCursor> {
-      --
-      -- We throw away nulls because there is a lot of wierdness when they
-      -- get included.
-      where
-        ${sql.join(pgAttributes.map(pgAttribute => sql.query`${sql.identifier(pgAttribute.name)} is not null`), ' and ')} and
-        ${beforeCursor ? this._getCursorCondition(pgAttributes, beforeCursor, descending ? '>' : '<') : sql.raw('true')} and
-        ${afterCursor ? this._getCursorCondition(pgAttributes, afterCursor, descending ? '<' : '>') : sql.raw('true')} and
-        ${conditionSql}
-
-      -- Order using the same attributes used to construct the cursors. If
-      -- a last property was defined we need to reverse our ordering so the
-      -- limit will work. We will fix the order in JavaScript.
-      order by ${sql.join(pgAttributes.map(pgAttribute =>
-        sql.query`${sql.identifier(pgAttribute.name)} using ${sql.raw((last != null ? !descending : descending) ? '>' : '<')}`,
-      ), ', ')}
-
-      -- Finally, apply the appropriate limit.
-      limit ${first != null ? sql.value(first) : last != null ? sql.value(last) : sql.raw('all')}
-
-      -- If we have an offset, add that as well.
-      ${_offset != null ? sql.query`offset ${sql.value(_offset)}` : sql.query``}
+      with ${sql.identifier(aliasIdentifier2)} as (
+        -- The standard select/from clauses up top.
+        select ${sql.identifier(aliasIdentifier)}.*
+        from ${fromSql} as ${sql.identifier(aliasIdentifier)}
+
+        -- Combine our cursors with the condition used for this page to
+        -- implement a where condition which will filter what we want it to.
+        --
+        -- We throw away nulls because there is a lot of wierdness when they
+        -- get included.
+        where
+          ${sql.join(pgAttributes.map(pgAttribute => sql.query`${sql.identifier(pgAttribute.name)} is not null`), ' and ')} and
+          ${beforeCursor ? this._getCursorCondition(pgAttributes, beforeCursor, descending ? '>' : '<') : sql.raw('true')} and
+          ${afterCursor ? this._getCursorCondition(pgAttributes, afterCursor, descending ? '<' : '>') : sql.raw('true')} and
+          ${conditionSql}
+
+        -- Order using the same attributes used to construct the cursors. If
+        -- a last property was defined we need to reverse our ordering so the
+        -- limit will work. We will fix the order in JavaScript.
+        order by ${sql.join(pgAttributes.map(pgAttribute =>
+          sql.query`${sql.identifier(pgAttribute.name)} using ${sql.raw((last != null ? !descending : descending) ? '>' : '<')}`,
+        ), ', ')}
+
+        -- Finally, apply the appropriate limit.
+        limit ${first != null ? sql.value(first) : last != null ? sql.value(last) : sql.raw('all')}
+
+        -- If we have an offset, add that as well.
+        ${_offset != null ? sql.query`offset ${sql.value(_offset)}` : sql.query``}
+      ) select to_json(${sql.identifier(aliasIdentifier2)}) as value from ${sql.identifier(aliasIdentifier2)};
```

Comment:
Could we add a comment here explaining why we have the `WITH` clause? My fear is that in the future we will be looking to migrate away from `to_json` (which could be sooner rather then later with https://github.com/calebmer/postgraphql/pull/342!) and we won’t know why the `WITH` is hear and we’ll just keep the `WITH`. It also helps to explain to future contributors why it is there.